web/src/app/schedules/calendar/ScheduleCalendar.tsx

```diff
@@ -123,81 +182,90 @@ function ScheduleCalendar(props) {
     )
   }
 
-  const getCalEvents = (shifts, _tempScheds, userOverrides) => {
-    const tempSchedules = _tempScheds.map((sched) => ({
-      start: sched.start,
-      end: sched.end,
+  const getCalEvents = (
+    shifts: OnCallShift[],
+    _tempScheds: TemporarySchedule[],
+    userOverrides: UserOverride[],
+  ): ScheduleCalendarEvent[] => {
+    const tempSchedules: TempSchedEvent[] = _tempScheds.map((sched) => ({
+      type: 'tempSched',
+      start: new Date(sched.start),
+      end: new Date(sched.end),
```

Comment:
Yeah so that `new Date()` code here was in the original JavaScript version of these files. You can find it in line 192 of the original. The old PR essentially converted all these shifts to one format then converted them again into a generic format at the end of the getCalEvents function. When I converted it to typescript, it made more sense not to have that intermediary conversion step, and instead just to convert everything to the output format right away. Because of that I moved the new Date code from the end of the getCalEvents function to instead being in the mapping for each of the events individually.

I agree, I actually backtracked some changes to try and make this PR have less changes compared to it's JavaScript version. If you want I can backtrack some more and bring back the original format of the getCalEvents function then make a follow-up PR to bring back cleaner typings. Let me know if that would make it easier!

Reply:
I think the idea was to keep everything as an ISOTimeStamp since it's ultimately just a string that can be parsed easily everywhere, but the calendar has (or at least had at one point) a hard requirement to use Javascript Date objects so we would convert when we want to render that component

Reply:
Checking out what the Cal is doing in master, I see what you're saying about the string vs date thing. I didn't realize we were setting it as a string then converting it to a Date all within the same function, with nothing really happening in between, so that seems fine to me. I've made a couple updates though (since I know you're on a new team now) so we can try and get this merged soon